src/h5web/visualizations/nexus/hooks.ts

```diff
@@ -51,14 +45,8 @@ export function useLinkedDatasetValues(
 export function useNxData(group: HDF5Group, metadata: HDF5Metadata): NxData {
   const datasetValues = useLinkedDatasetValues(group, metadata);
 
-  const signalName = getAttributeValue(group, 'signal');
-  assertStr(signalName);
-
-  const signalDataset = getLinkedEntity(signalName, group, metadata);
-  assertDefined(signalDataset, 'Signal dataset not found');
-  assertDataset(signalDataset);
-  assertNumericType(signalDataset);
-  assertSimpleShape(signalDataset);
+  const signalName = findSignalName(group);
+  const signalDataset = findSignalDataset(signalName, group, metadata);
```

Comment:
Perhaps it would be more handy to have a single function that returns the name and the dataset ?

In the end, `findSignalName` and `findSignalDataset` will not be used on their own.

Reply:
Or, we could imagine using `id` instead of names to index `datasetValues`

Reply:
Great idea!

Reply:
I've done some refactoring, but I'll open a new PR.